src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseFileRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ReleaseFileRepository : IReleaseFileRepository
+    {
+        private readonly ContentDbContext _contentDbContext;
+
+        public ReleaseFileRepository(ContentDbContext contentDbContext)
+        {
+            _contentDbContext = contentDbContext;
+        }
+
+        public async Task<Either<ActionResult, ReleaseFileReference>> CheckFileExists(Guid releaseId, Guid id,
+            params ReleaseFileTypes[] allowedFileTypes)
+        {
+            // Ensure file is linked to the Release by getting the ReleaseFile first
+            var releaseFile = await Get(releaseId, id);
+
+            if (releaseFile == null)
+            {
+                return new NotFoundResult();
+            }
+
+            if (allowedFileTypes.Any() && !allowedFileTypes.Contains(releaseFile.ReleaseFileReference.ReleaseFileType))
+            {
+                return ValidationUtils.ValidationActionResult(FileTypeInvalid);
+            }
+
+            return releaseFile.ReleaseFileReference;
+        }
+
+        public async Task Delete(Guid releaseId, Guid fileId)
+        {
+            var releaseFile = await Get(releaseId, fileId);
+            if (releaseFile != null)
+            {
+                _contentDbContext.ReleaseFiles.Remove(releaseFile);    
+            }
```

Comment:
Shouldn't we save the context here too?

Reply:
I didn't save in these methods on purpose here at the lower repository level as the caller might want to make a group of changes across multiple methods, possible across repositories or mixed with other changes using the database context directly in the service, and persist all of them together in single save, or roll back.

Reply:
Hmm... Not sure this is a good idea for a few reasons:

1. Puts the onus on the consumers to save the changes. This is a potential gotcha as one could reasonably expect that repository methods to do so without having to intimately know what the code is doing. This case is similar to how the `FootnoteService` from my other PR didn't actually save when deleting footnotes.
2. Leaks implementation details as consumers shouldn't need to know what context the repository is using under-the-hood. Imagine if we wanted to change the underlying implementation to some other context (not that we would), we would also need to update all the various call sites too.

From an efficiency perspective, I think it's really unlikely there'd be any measurable performance hit from performing additional saves (particularly as these won't be high traffic parts of the application).

From a rollback perspective, if we were intending to roll it back, there's multiple things we could do: 
- Set up a transaction within the service (leaks implementation details, but fine for this particular purpose)
- Use the context directly and forego repository methods
- Add more repository methods that handle that kind of more complex logic

I think these kinds of things make me appreciate Spring's transaction boundaries a bit more, although that's not without its own problems...

Reply:
I have made this change! I was more worried about the rollback perspective, where we don't want to end up with say, a data file reference without the corresponding meta file reference. I guess with very simple logic like we have there is very little to go wrong but in complicated situations we can think more carefully about the transaction.

Reply:
Cool. Yep indeed, I'd probably recommend using the context directly in those kinds of instances!